consensus/rpcclient/client.go

```diff
@@ -0,0 +1,100 @@
+package consensusrpcclient
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/node"
+
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/consensus"
+	"github.com/offchainlabs/nitro/execution"
+	"github.com/offchainlabs/nitro/util/containers"
+	"github.com/offchainlabs/nitro/util/rpcclient"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+type ConsensusRPCClient struct {
+	stopwaiter.StopWaiter
+	client *rpcclient.RpcClient
+}
+
+func NewConsensusRpcClient(config rpcclient.ClientConfigFetcher, stack *node.Node) *ConsensusRPCClient {
+	return &ConsensusRPCClient{
+		client: rpcclient.NewRpcClient(config, stack),
+	}
+}
+
+func (c *ConsensusRPCClient) Start(ctx_in context.Context) error {
+	c.StopWaiter.Start(ctx_in, c)
+	ctx := c.GetContext()
+	return c.client.Start(ctx)
+}
+
+func convertError(err error) error {
+	if err == nil {
+		return nil
+	}
+	errStr := err.Error()
+	if strings.Contains(errStr, execution.ErrSequencerInsertLockTaken.Error()) {
+		return execution.ErrSequencerInsertLockTaken
+	}
+	return err
+}
+
+func (c *ConsensusRPCClient) FindInboxBatchContainingMessage(message arbutil.MessageIndex) containers.PromiseInterface[consensus.InboxBatch] {
+	return stopwaiter.LaunchPromiseThread(c, func(ctx context.Context) (consensus.InboxBatch, error) {
+		var res consensus.InboxBatch
+		err := c.client.CallContext(ctx, &res, consensus.RPCNamespace+"_findInboxBatchContainingMessage", message)
+		if err != nil {
+			return consensus.InboxBatch{BatchNum: 0, Found: false}, convertError(err)
+		}
+		return res, nil
```

Comment:
This same pattern can also be applied to the other ConsensusRPCClient funcs
```suggestion
		return res, convertError(err)
```